var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -533,7 +555,12 @@ class Chapel(AutotoolsPackage, CudaPackage, ROCmPackage):
     depends_on("gasnet conduits=none", when="gasnet=spack")
     depends_on("gasnet@2024.5.0: conduits=none", when="@2.1.0: gasnet=spack")
 
+    with when("+python-bindings"):
+        extends("python")
+        requires("%clang", "%apple-clang", policy="one_of")
```

Comment:
I don't understand this `requires` clause. This goes far beyond simply saying that Chapel must be using LLVM (which also generally sets `CHPL_TARGET_{CC,CXX}` to LLVM), it additionally requires that the toolchain used to _build that LLVM_ must also be LLVM (and not for example GCC used to build LLVM).

Is this an actual requirement for correctness?

Reply:
no, that's not a correctness requirement, and I think is actually solved with the previous suggestion to combine the `llvm=none` conflict.